src/wallet/rpcwallet.cpp

```diff
@@ -3478,19 +3479,76 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "From address does not belong to this node, zaddr spending key or viewing key not found.");
     }
 
-    UniValue result(UniValue::VARR);
     std::vector<SproutNoteEntry> sproutEntries;
     std::vector<SaplingNoteEntry> saplingEntries;
     auto noteFilter = AddrSet::ForPaymentAddresses(std::vector({decoded.value()}));
     pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, noteFilter, nMinDepth, INT_MAX, false, false);
 
+    auto push_transparent_result = [&](const CTxDestination& dest) -> void {
+        bool haveKey = std::visit(match {
+            [&](const CKeyID& key) { return pwalletMain->HaveKey(key); },
+            [&](const CScriptID& script) { return pwalletMain->HaveCScript(script); },
+            [&](const auto& other) { return false; }
+        }, dest);
+        const CScript scriptPubKey{GetScriptForDestination(dest)};
+        for (const auto& [_txid, wtx] : pwalletMain->mapWallet) {
+            if (wtx.IsCoinBase() || !CheckFinalTx(wtx))
+                continue;
+
+            int nDepth = wtx.GetDepthInMainChain();
+            if (nDepth < nMinDepth) continue;
+            for (size_t i = 0; i < wtx.vout.size(); ++i) {
+                const CTxOut& txout{wtx.vout[i]};
+                if (txout.scriptPubKey == scriptPubKey) {
+                    UniValue obj(UniValue::VOBJ);
+                    obj.pushKV("pool", "transparent");
+                    obj.pushKV("txid", wtx.GetHash().ToString());
+                    obj.pushKV("amount", ValueFromAmount(txout.nValue));
+                    obj.pushKV("amountZat", txout.nValue);
+                    obj.pushKV("outindex", int(i));
```

Comment:
That's a good idea; I thought `actionindex` is useful so the caller could distinguish between sapling and orchard entries, but now that we have the `pool` (or `type`) field, it's better to use a consistent name, as  you suggest. Unless you think otherwise, I won't try to fix the other APIs in this PR; we can open a separate PR for those.